docs/topics/Opcodes_and_Status_Codes.md

```diff
@@ -182,6 +182,9 @@ Along with the HTTP error code, our API can also return more detailed error code
 | 50041  | Invalid API version provided                                                                                                  |
 | 50074  | Cannot delete a channel required for Community guilds                                                                         |
 | 50081  | Invalid sticker sent                                                                                                          |
+| 50083  | Invalid thread archive state (e.g. Tried to archive an archived thread)                                                       |
```

Comment:
Can't this be idempotent instead? 🙏

Reply:
Or is this also the error given if you try to unarchive an archived threads when it is `locked`?

Reply:
Haven't built locking yet, so not sure if that will use the same error code.

What's the reason for it to be idempotent instead of returning an error?

Reply:
I believe a reason might be not having to keep track of state, for example if you schedule some sort of "archiving" in a bot and just toss the task into a task queue. Then, when the task gets run, it could just archive the thread instead of asking the cache what the thread is currently like.

Reply:
> What's the reason for it to be idempotent instead of returning an error?

Other endpoints are idempotent, from my knowledge. If you pin an already pinned message there's no error, if you mute/deafen a member that's already muted/deafened there's no error, if you toggle NSFW on an already NSFW channel there's no error, etc.

Reply:
+1 to this, we try not to punish clients too much for not keeping their own state of the discord world.

Reply:
Alright, I'll make it idempotent, thanks